pkg/cloud_provider/file/file.go

```diff
@@ -681,15 +681,22 @@ func IsContextError(err error) *codes.Code {
 	return nil
 }
 
-// PollOpErrorCode returns a pointer to the grpc error code that maps to the http
-// error code for passed in googleapi error. Returns grpc DeadlineExceeded if the
-// given error contains the "context deadline exceeded" string. Returns the grpc Internal error
-// code if the passed in error is neither a user error or a deadline exceeded error.
-func PollOpErrorCode(err error) *codes.Code {
-	if errCode := IsUserError(err); errCode != nil {
+// CodeForError returns a pointer to the grpc error code that maps to the http
+// error code for the passed in user googleapi error or context error. Returns
+// codes.Internal if the given error is not a googleapi error caused by the user.
+// The following http error codes are considered user errors:
+// (1) http 400 Bad Request, returns grpc InvalidArgument,
+// (2) http 403 Forbidden, returns grpc PermissionDenied,
+// (3) http 404 Not Found, returns grpc NotFound
+// (4) http 429 Too Many Requests, returns grpc ResourceExhausted
+// The following errors are considered context errors:
+// (1) "context deadline exceeded", returns grpc DeadlineExceeded,
+// (2) "context canceled", returns grpc Canceled
+func CodeForError(err error) *codes.Code {
+	if errCode := isUserError(err); errCode != nil {
 		return errCode
 	}
-	if errCode := IsContextError(err); errCode != nil {
+	if errCode := isContextError(err); errCode != nil {
 		return errCode
 	}
 	return util.ErrCodePtr(codes.Internal)
```

Comment:
Can we also check if there is already some known status.code set in the err and skip overwrite of code to Internal.

if s, ok := status.FromError(test.err); ok {
 ... if s.Code() == // one of [codes](https://github.com/grpc/grpc-go/blob/v1.55.0/codes/codes.go)
}

Reply:
Done!